gap/attributes/attr.gi

```diff
@@ -1045,13 +1045,173 @@ function(S)
 
   return
     function(x, y)
-      local R;
-      R := RClass(S, x);
-      return IsGreensLessThanOrEqual(R, RClass(S, y))
-        and ForAny(Idempotents(R), e -> e * y = x);
+      local E;
+      E := Idempotents(S);
+      return ForAny(E, e -> e * y = x)
+        and ForAny(E, f -> y * f = x);
     end;
 end);
 
+SEMIGROUPS.ExistsTransversal := function(a, b, n)
+    local exists, class, i;
+    if DegreeOfTransformation(a) > n or
+        DegreeOfTransformation(b) > n then
+        ErrorNoReturn("Transformation does not",
+        "belong to a transformation semigroup of degree n");
+    fi;
+    for class in KernelOfTransformation(a, n) do
+        exists := false;
+        for i in [1 .. Size(class)] do
+            if class[i] ^ a = class[i] ^ b then
+                exists := true;
+                break;
+            fi;
+        od;
+        if not exists then
+            return false;
+        fi;
```

Comment:
Not saying this is better, but I think you could also write it like this (assuming `class` is an iterable object):
```suggestion
        if ForAll(class, x -> x ^ a <> x ^ b) then
            return false;
        fi;
```